mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -590,7 +590,7 @@ private List<MailboxRenamedResult> doRenameMailbox(Mailbox mailbox, MailboxPath
                 .asUserBound();
             locker.executeWithLock(from, (LockAwareExecution<Void>) () -> {
                 block(mapper.findMailboxWithPathLike(query)
-                    .flatMap(sub -> {
+                    .concatMap(sub -> {
```

Comment:
I think we can agree this concatMap is not really beneficial...

Should we roll it back? What do you think?

Reply:
Me I think RENAME is not a common command to be called much in the real world and even its edge case with so many sub mailboxes, therefore I would agree on choosing the performance here.